### PR TITLE
UI: introduce detached notification provider

### PR DESCRIPTION
**Pull request by Undin**
Extract "File is not included in module tree, ..." notification introduced in #3826 to separate notification provider and allow to disable such notifications (https://github.com/intellij-rust/intellij-rust/issues/771#issuecomment-496451670)

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/notifications/RsNotificationProvider.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.ui.EditorNotificationPanel
+import com.intellij.ui.EditorNotifications
+
+abstract class RsNotificationProvider(
+    protected val project: Project
+) : EditorNotifications.Provider<EditorNotificationPanel>() {
+
+    protected abstract val disablingKey: String
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/notifications/RsNotificationProvider.kt]: Consider moving it to the constructor. (and remove redundant `const`s)

Comment (Undin) [reply src/main/kotlin/org/rust/ide/notifications/RsNotificationProvider.kt]: I've rewritten implementation as we had discussed in person.
Now it's possible to provide `disablingKey` per file. So I'd prefer not to pass any constant into the constructor to keep base implementation the most flexible

**Comment by vlad20012** [comment]
bors r_

**Comment by mchernyavsky** [comment]
bors retry

**Pull request merged**